### PR TITLE
Ability to specify db config in import tool

### PR DESCRIPTION
**Pull request by tinwelint**
so that e.g array_block_size and string_block_size may be specified
along with creating the store.

### CODE REVIEW

File: community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java [hunk 1]
```diff
@@ -181,7 +183,11 @@
                 "<true/false>",
                 "Whether or not to ignore extra columns in the data not specified by the header. "
                         + "Skipped columns will be logged, containing at most number of entities specified by "
-                        + BAD_TOLERANCE.key() + "." );
+                        + BAD_TOLERANCE.key() + "." ),
+        DATABASE_CONFIG( "db-config", null,
+                "<path/to/neo4j.properties>",
+                "(advanced) File specifying database-specific configuration. For more information consult "
+                        + "manual about available configuration options for a neo4j configuration file" );
```
Comment (jexp) [inline community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java]: perhaps we should explicitly call out the block sizes?
We should probably also note that _only store-structure related_ properties are used? e..g NOT dbms.pagecache.memory or keep_logical_logs or legacy-auto-index config.

File: community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java [hunk 2]
```diff
@@ -1176,6 +1182,30 @@ public void shouldBeEquivalentToUseRawAsciiOrCharacterAsQuoteConfiguration2() th
         }
     }
 
+    @Test
+    public void shouldRespectDbConfig() throws Exception
+    {
+        // GIVEN
+        int size = 10;
```
Comment (jexp) [inline community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java]: can we use two different sizes here, just to make sure they land up individually correctly?

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java [hunk 3]
```diff
@@ -84,12 +84,12 @@
...
     public BatchingNeoStores( FileSystemAbstraction fileSystem, File storeDir, Configuration config,
-            LogService logService, AdditionalInitialIds initialIds )
+            LogService logService, AdditionalInitialIds initialIds, Config dbConfig )
     {
         this.fileSystem = fileSystem;
         this.logProvider = logService.getInternalLogProvider();
         this.storeDir = storeDir;
-        this.neo4jConfig = new Config( stringMap(
+        this.neo4jConfig = new Config( stringMap( dbConfig.getParams(),
                 dense_node_threshold.name(), valueOf( config.denseNodeThreshold() ),
```
Comment (jexp) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java:93]: should we have the passed in db-config have priority e.g. over dense-node-threshold as that is a store-related property? (not sure about page-cache size)

**Comment by jexp** [comment]
@tinwelint had some smallish comments, code-wise it looks good to me.

**Comment by spacecowboy** [comment]
@tinwelint needs rebase

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java (hunk 3 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java:93]: It's done now (when I push) where the threshold from the dbConfig is used to override the `Configuration` object passed into `BatchingNeoStores`... that will have it put that back into Config, but that's fine and mostly an inconvenience in the import tool, not so for direct ParallelBatchImporter usage.

**Comment by tinwelint** [comment]
@jexp have a look again

**Pull request merged**